Add new `Layout` design page

yurm04 opened the pull request:
<!--
  ☝️How to write a good PR title:
  - Prefix it with [ComponentName] (if applicable), for example: [Button]
  - Start with a verb, for example: Add, Delete, Improve, Fix…
  - Give as much context as necessary and as little as possible
  - Prefix it with [WIP] while it’s a work in progress
-->

### WHY are these changes introduced?

Fixes #8561  <!-- link to issue if one exists -->

<!--
  Context about the problem that’s being addressed.
-->

### WHAT is this pull request doing?
Adding a new `Layout` resource page to the `Design` section in the polaris.shopify.com site
![localhost_3000_design_layout (1)](https://user-images.githubusercontent.com/4642404/228296784-403fb0f3-1982-46c9-8a32-75268f02d5b3.png)

<!--
  Summary of the changes committed.

  Before / after screenshots are appreciated for UI changes. Make sure to include alt text that describes the screenshot.

  If you include an animated gif showing your change, wrapping it in a details tag is recommended. Gifs usually autoplay, which can cause accessibility issues for people reviewing your PR:

    <details>
      <summary>Summary of your gif(s)</summary>
      <img src="..." alt="Description of what the gif shows">
    </details>
-->

<!-- ℹ️ Delete the following for small / trivial changes -->

### How to 🎩

🖥 [Local development instructions](https://github.com/Shopify/polaris/blob/main/README.md#local-development)
🗒 [General tophatting guidelines](https://github.com/Shopify/polaris/blob/main/documentation/Tophatting.md)
📄 [Changelog guidelines](https://github.com/Shopify/polaris/blob/main/.github/CONTRIBUTING.md#changelog)

<!--
  Give as much information as needed to experiment with the component
  in the playground.
-->

<details>
<summary>Copy-paste this code in <code>playground/Playground.tsx</code>:</summary>

```jsx
import React from 'react';
import {Page} from '../src';

export function Playground() {
  return (
    <Page title="Playground">
      {/* Add the code you want to test in here */}
    </Page>
  );
}
```

</details>

### 🎩 checklist

- [ ] Tested on [mobile](https://github.com/Shopify/polaris/blob/main/documentation/Tophatting.md#cross-browser-testing)
- [ ] Tested on [multiple browsers](https://help.shopify.com/en/manual/shopify-admin/supported-browsers)
- [ ] Tested for [accessibility](https://github.com/Shopify/polaris/blob/main/documentation/Accessibility%20testing.md)
- [ ] Updated the component's `README.md` with documentation changes
- [ ] [Tophatted documentation](https://github.com/Shopify/polaris/blob/main/documentation/Tophatting%20documentation.md) changes in the style guide

yurm04 committed:
added new layout design page

yurm04 committed:
add sidebyside

yurm04 committed:
add side by side content and assets

yurm04 committed:
Merge branch 'main' of https://github.com/Shopify/polaris into ye/add-layout-page

yurm04 committed:
added example images and sandbox links

tracytees committed:
removed old images and added @2x versions as well as new banner image

jjgali committed:
Content edits

yurm04 committed:
updated content, fixed banner styleing, added alt text for images and video

yurm04 commented on polaris.shopify.com/content/design/layout.md:
```diff
@@ -0,0 +1,92 @@
+  <img src="/images/foundations/design/layout/layout-banner.png" alt="A visual example of a polaris unit">
+</div> -->
+
+![Building blocks in various combinations coming together to make up the same overall structure](/images/foundations/design/layout/layout-banner.png)
```
@jjgali mind giving all of the alt text an edit pass? 🙏

yurm04 commented on polaris.shopify.com/content/design/layout.md:
```diff
@@ -0,0 +1,92 @@
+
+## Layout component examples
+
+The following examples illustrate how the new layout components can be composed together to create admin UI elements. Each example is just one approach to building that particular bit of UI, but many other combinations of layout components can result in the same outcome. To get a feel for these new components, try to rebuild the same UI using a different combination of our new layout components.
```
@jjgali added this blurb here to try and make clearer what these examples were for. Please feel free to edit it, I kind of wrote it on the fly last minute

yurm04 committed:
changeset

jjgali committed:
Content edits

Since we aren't including the Figma file link right now, the designing section was pretty sparse. I brought `Breakpoints and spacing` up to an h2, and moved the line about browsing components to the top section of the page.

We can consider restructuring this as more assets like the Figma file become available.

laurkim commented on polaris.shopify.com/content/design/layout.md:
```diff
@@ -0,0 +1,87 @@
+  - columns
+  - bleed
+  - card
+  - Box
```
```suggestion
  - box
  - inline
  - divider
  - responsive
  - spacing
```

jjgali committed:
Alt text revisions

- Remove alt text for decorative image
- Move example descriptions out of alt text and into main body content so it can be read with or without a screen reader

jjgali committed:
Update polaris.shopify.com/content/design/layout.md

Co-authored-by: Lo Kim <lo.kim@shopify.com>

jjgali reviewed (approved):
LGTM! Content review finished.

laurkim reviewed (approved):
Nice!! 💯 🥇 :shipit:
